src/main/java/seedu/address/logic/commands/paymentcommand/PaymentReceiveCommand.java

```diff
@@ -71,23 +75,25 @@ public CommandResult execute(Model model) throws CommandException {
         Payment existingPayment = tuteeToGet.getPayment();
         String existingPaymentValue = existingPayment.getValue();
         LocalDate existingPayByDate = existingPayment.getPayByDate();
-        Tutee editedTutee;
 
-        if (zeroPaymentVal.equals(existingPaymentValue) && newPayByDate == null && existingPayByDate == null) {
+        boolean hasZeroPaymentValue = ZERO_PAYMENT_VAL.equals(existingPaymentValue);
```

Comment:
As "0" is not equal to "0.00", perhaps a simpler implementation would be 
```suggestion
        boolean hasZeroPaymentValue = Double.parseDouble(existingPaymentValue) == 0;
```

Reply:
Thanks! Have committed the above suggestions